BigQuery/tests/System/ManageTablesTest.php

```diff
@@ -255,6 +255,79 @@ public function testIam()
         $this->assertEquals([$perm], $iam->testPermissions([$perm]));
     }
 
+    public function testSnapshot()
+    {
+        $destinationTable = self::$dataset->table(uniqid('test_dest_table_'));
+        $copyConfig = self::$table->copy(
+            $destinationTable,
+            ['configuration' => ['copy' => ['operationType' => 'SNAPSHOT']]],
+        );
+        $this->runJob($copyConfig);
+        $snapshotDefinition = $destinationTable->reload()['snapshotDefinition'];
+
+        // Assert for proper base table reference
+        $this->assertEquals(
+            $snapshotDefinition['baseTableReference'],
+            self::$table->info()['tableReference']
+        );
+
+        // Assert for proper RFC3339 extended format timestamp
+        $isRfc3339_extended = \DateTime::createFromFormat(
+            \DateTime::RFC3339_EXTENDED,
+            $snapshotDefinition['snapshotTime']
+        ) === false ? false : true;
+        $this->assertTrue($isRfc3339_extended);
+
+        // Test snapshot restore operation
+        $restoredTable = self::$dataset->table(uniqid('restored_table_'));
+        $copyConfig = $destinationTable->copy(
+            $restoredTable,
+            ['configuration' => ['copy' => ['operationType' => 'RESTORE']]],
+        );
+        $this->runJob($copyConfig);
+    }
+
+    public function testCreateClone()
```

Comment:
Test data providers should not start with name `test`, else phpunit thinks its also a test method.
Maybe rename to `tableCloneProvider `

Reply:
No, it is a test in itself. It tests if clone if getting created properly.

Reply:
Also, since this method doesn't actually provide multiple values, you are better off with a private helper method.

Reply:
Got it.